CRM/Pledge/BAO/PledgePayment.php

```diff
@@ -602,34 +615,44 @@ public static function calculateNextScheduledDate(&$params, $paymentNo, $basePay
    *   $statusId calculated status id of pledge
    */
   public static function calculatePledgeStatus($pledgeId) {
-    $paymentStatusTypes = CRM_Contribute_PseudoConstant::contributionStatus(NULL, 'name');
+    $contributionStatuses = CRM_Core_PseudoConstant::get('CRM_Contribute_DAO_Contribution', 'contribution_status_id', array(
```

Comment:
If you look at the function it says that is is supposed to be internal & you should use CRM_Contribute_BAO_Contribution::buildOptions('contribution_status_id');

But there is something else going on here too. For pledges you should use the pledge bao & for pledge payments the pledge payment bao. They all use the same option group at the moment - contribution_status. But really they should not & in fact the pledge buildOptions functions removes Failed from the options. The 'right' answer is to split them out into separate option groups I think since 'In Progress' is pledge-only whereas 'Refunded' is contribution only.  That is out of scope, but using the right BAO for the fn call now will make that easier later